zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/changes/PartitionDisableExporterApplier.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.dynamic.config.changes;
+
+import io.atomix.cluster.MemberId;
+import io.camunda.zeebe.dynamic.config.changes.ConfigurationChangeAppliers.MemberOperationApplier;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfiguration;
+import io.camunda.zeebe.dynamic.config.state.DynamicPartitionConfig;
+import io.camunda.zeebe.dynamic.config.state.ExporterState;
+import io.camunda.zeebe.dynamic.config.state.ExporterState.State;
+import io.camunda.zeebe.dynamic.config.state.MemberState;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.scheduler.future.CompletableActorFuture;
+import io.camunda.zeebe.util.Either;
+import java.util.function.UnaryOperator;
+
+final class PartitionDisableExporterApplier implements MemberOperationApplier {
+
+  private final int partitionId;
+  private final MemberId memberId;
+  private final String exporterId;
+  private final PartitionChangeExecutor partitionChangeExecutor;
+
+  public PartitionDisableExporterApplier(
+      final int partitionId,
+      final MemberId memberId,
+      final String exporterId,
+      final PartitionChangeExecutor partitionChangeExecutor) {
+    this.partitionId = partitionId;
+    this.memberId = memberId;
+    this.exporterId = exporterId;
+    this.partitionChangeExecutor = partitionChangeExecutor;
+  }
+
+  @Override
+  public MemberId memberId() {
+    return memberId;
+  }
+
+  @Override
+  public Either<Exception, UnaryOperator<MemberState>> initMemberState(
+      final ClusterConfiguration currentClusterConfiguration) {
+
+    if (!currentClusterConfiguration.hasMember(memberId)) {
+      return Either.left(
+          new IllegalStateException(
+              String.format(
+                  "Expected to disable exporter, but the member '%s' does not exist in the cluster",
+                  memberId)));
+    }
+
+    final MemberState member = currentClusterConfiguration.getMember(memberId);
+    final var partitionExistsInLocalMember = member.hasPartition(partitionId);
+
+    if (!partitionExistsInLocalMember) {
+      return Either.left(
+          new IllegalStateException(
+              String.format(
+                  "Expected to disable exporter, but the member '%s' does not have the partition '%s'",
+                  memberId, partitionId)));
+    }
+
+    final var partitionHasExporter =
+        member.getPartition(partitionId).config().exporting().exporters().containsKey(exporterId);
+
+    if (!partitionHasExporter) {
```

Comment:
> if the exporter doesn't exist, then there was a misconfiguration?

This can happen usually when the user misspelled the `exporterId` in the disable request. So it makes sense to fail the request instead so that user can fix it a retry the request. 

Note that the request fails before actually starting to apply the changes because we have a validation step in the `ConfigurationChangeCoordinatorImpl` where it simulates applying the operations. So it won't fail after we start applying changes.